spark3/parquet-compression.choose-types-wisely.scala

```diff
@@ -0,0 +1,118 @@
+
+// our initial data is generated as one million random numbers.
+val rndDF = spark.sparkContext
+  .parallelize(
+    List.range (0 ,1000000)
+      .map(I=>(math.random * 1000000)
+      ) )
+      .toDF
+
+
+// COMMAND ----------
+println (s"${"-"*20}")
+println (s"Write data")
+
+// let's write the data using all expected types and compressions
+for (numericTypeName <- allNumericTypes) {
+  for (parquetCompressionName <- allParquetCompressions ){
+    val fileName=s"1M/Parquet_${numericTypeName}_${parquetCompressionName}"
+    print (".")
+    spark.sparkContext.setJobGroup("Write data",s"Write ${numericTypeName} with compression ${parquetCompressionName}")
+    rndDF.selectExpr(s"cast(value as $numericTypeName )")
+      .write
+      .option("compression", parquetCompressionName)
+      .format("parquet")
+      .mode("overwrite")
+      .save(fileName)
+
+  }
+}
+
+
+// COMMAND ----------
+
+// Now it's up to you to manually check the amount of storage required for each generated folder ...
+
+// ... or to use Spark's Hadoop configuration to do it for you:
+println (s"${"-"*20}")
+println (s"Check written data size")
+val hadoopFileSystem = org.apache.hadoop.fs.FileSystem.get(
+  java.net.URI.create("./")
+  , spark.sparkContext.hadoopConfiguration)
+
+//search for sum of files size in kiloBytes, per parent folder:
+val sizeOnDisk= hadoopFileSystem.globStatus(new org.apache.hadoop.fs.Path("./1M/**/part*"))
+  .filter(o=>o.isFile)
+  .map(o=> (o.getLen/1024, o.getPath.getParent.toString))
+  .groupBy(_._2)
+  .mapValues(_.map(_._1).sum).toSeq.sortBy(_._2)
+
+println("part* files sizes (in kB):")
+sizeOnDisk.foreach( o=>println ( s"${o._1}\t${o._2}\tkB"))
+
+// COMMAND ----------
+
+// now we can also add a check on the effect of choosing a specific number type on the obtained values
+// and compression on the obtained files size.
+println (s"${"-"*20}")
+println (s"Read written data")
+import scala.collection.mutable.ArrayBuffer
+val readInformations: ArrayBuffer[(String,Long,Any)] = ArrayBuffer.empty
+for (numericTypeName <- allNumericTypes) {
+  for (parquetCompressionName <- allParquetCompressions ){
+    val fileName=s"1M/Parquet_${numericTypeName}_${parquetCompressionName}"
+    print(".")
+    spark.sparkContext.setJobGroup("Read data",s"Read ${numericTypeName} with compression ${parquetCompressionName}")
+
+    val t0=System.nanoTime
+    val aggValue = spark.read.parquet(fileName)
+      .selectExpr (s"avg(value) as `avg_from_${numericTypeName}_${parquetCompressionName}`")
+      .first
+      .get(0)
+
+    val t1=System.nanoTime
+    readInformations.append((fileName,(t1-t0)/1000000, aggValue))
+  }}
+
+
+println("Time to read and calculating aggregated value")
+readInformations.foreach(readInformation=> println ( s"${readInformation._1}\t${readInformation._2} ms\t(agg value was ${readInformation._3})" ))
+
+
+
+// COMMAND ----------
+
+// Now you should also check how the variety of values affects compression :
```

Comment:
I'd encourage you to add some conclusions here so that people reading the snippet can take action and understand what to expect with their change.